### PR TITLE
Storage: Import Support

### LINKED ISSUE #1302: No import facility for storage containers?
Issues on GitHub are intended to be related to bugs or feature requests with provider codebase,
so we recommend using our other community resources instead of asking here 👍.

---

If you have a support request or question please submit them to one of these resources:

* [Terraform community resources](https://www.terraform.io/docs/extend/community/index.html)
* [HashiCorp support](https://support.hashicorp.com) (Terraform Enterprise customers)

I see that I can import an existing storage account https://www.terraform.io/docs/providers/azurerm/r/storage_account.html but there's no corresponding documentation for https://www.terraform.io/docs/providers/azurerm/r/storage_container.html. 

Is this because containers are accessed on the Data Plane and not the Management Plane? 

/cc @rohrerb

### LINKED ISSUE #1554: azurerm_storage_container.NAME.id returns an unpathed resource ID
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.7
+ provider.azurerm v1.9.0
+ provider.random v1.3.1

### Affected Resource(s)

* azurerm_storage_container

### Terraform Configuration Files

```hcl
resource "azurerm_storage_container" "assets" {
  name                  = "assets"
  resource_group_name   = "${azurerm_resource_group.app.name}"
  storage_account_name  = "${azurerm_storage_account.app.name}"
  container_access_type = "blob" # anon read for blobs
}

resource "azurerm_role_assignment" "role_assignment" {
  scope                = "${azurerm_storage_container.assets.id}"
  role_definition_id   = "${azurerm_role_definition.blob_contributor.id}"
  principal_id         = "${local.principal_id}"

  lifecycle {
    ignore_changes = ["name"]
  }
}
```

### Debug Output

(available on request)

### Expected Behavior

`azurerm_storage_container.NAME.id` should return a complete path to the resource ID

### Actual Behavior

`azurerm_storage_container.NAME.id` returns only the storage container name:

```
Terraform will perform the following actions:

  + azurerm_role_assignment.role_assignment
      id:                 <computed>
      name:               <computed>
      principal_id:       "c0824800-21a1-471a-8863-xxxx"
      role_definition_id: "/subscriptions/16b9f5dc-3276-4b32-91b2-xxxx/providers/Microsoft.Authorization/roleDefinitions/29b53033-c386-ba60-e9c0-xxxx"
      scope:              "assets"
```

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

Ø

### References

Ø

### LINKED ISSUE #1830: Cannot import Storage Account container
How can I import a container I have previously created in a Storage Account?

The following link does not provide information on how to do this, unlike that of storage_account:

https://www.terraform.io/docs/providers/azurerm/r/storage_container.html

Thanks in advance!

### PR DESCRIPTION
**Pull request by tombuildsstuff**
This PR is a dependency for #1746 - and adds support for Import to the Storage Resources. In order to do this the ID's need changing to contain all of the necessary information - which requires a state migration, and will require appropriate changelog comments.

- [x] Import support for `azurerm_storage_blob`
- [x] Import support for `azurerm_storage_container`
- [x] Import support for `azurerm_storage_queue`
- [x] Import support for `azurerm_storage_table`

Fixes #1302
Fixes #1554
Fixes #1830

### CODE REVIEW

File: azurerm/resource_arm_storage_queue.go [hunk 1]
```diff
@@ -87,84 +93,122 @@ func resourceArmStorageQueueCreate(d *schema.ResourceData, meta interface{}) err
...
+	// https://myaccount.queue.core.windows.net/myqueue
+	uri, err := url.Parse(input)
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing %q as a URI: %+v", input, err)
+	}
+
+	segments := strings.Split(uri.Host, ".")
+	if len(segments) > 0 {
+		storageAccountName := segments[0]
+		// remove the
```
Comment (mbfrahry) [inline azurerm/resource_arm_storage_queue.go]: Is this an unfinished comment?

File: azurerm/resource_arm_storage_table.go [hunk 2]
```diff
@@ -75,79 +83,133 @@ func resourceArmStorageTableCreate(d *schema.ResourceData, meta interface{}) err
...
+	// https://myaccount.table.core.windows.net/table1
+	uri, err := url.Parse(input)
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing %q as a URI: %+v", input, err)
+	}
+
+	segments := strings.Split(uri.Host, ".")
+	if len(segments) > 0 {
+		storageAccountName := segments[0]
+		// remove the
```
Comment (mbfrahry) [inline azurerm/resource_arm_storage_table.go]: Same here

Comment (tombuildsstuff) [reply azurerm/resource_arm_storage_table.go]: Yep - good spot, ta

**Comment by tombuildsstuff** [comment]
Ignoring a couple of expected test failures (to do with a change within Azure where encrypted storage accounts are now mandatory) - the tests pass:

screenshot 2018-08-30 at 13 36 29

**Comment by tombuildsstuff** [comment]
* `azurerm_storage_blob|container|queue|table` - support for import [GH-1816]
* `azurerm_storage_blob|container|queue|table` - support for import [GH-1816]
* `azurerm_storage_blob|container|queue|table` - support for import [GH-1816]

**Pull request merged**